django_swagger_tester/utils.py

```diff
@@ -255,3 +296,113 @@ def type_placeholder_value(_type: str) -> Any:
+    """
+    new_dict = {}
+    for key, value in data.items():
+        if isinstance(key, str) and '_' in key:
+            new_key = re.sub(camelize_re, underscore_to_camel, key)
+        else:
+            new_key = key
+        new_dict[new_key] = value
+    return new_dict
+
+
+def safe_validate_response(response: Response, path: str, method: str, func_logger: Callable) -> None:
+    """
+    Validates an outgoing response object against the OpenAPI schema response documentation.
+
+    In case of inconsistencies, a log is logged at a setting-specified log level.
+
+    Unlike the django_swagger_tester.testing validate_response function,
+    this function should *not* raise any errors during runtime.
+
+    :param response: HTTP response
+    :param path: The request path
+    :param method: The request method
+    :param func_logger: A logger callable
+    """
+    logger.info('Validating response for %s request to %s', method, path)
+    from django_swagger_tester.configuration import settings
+
+    try:
+        # load the response schema
+        response_schema = settings.loader_class.get_response_schema_section(
+            route=path, status_code=response.status_code, method=method, skip_validation_warning=True,
+        )
+    except UndocumentedSchemaSectionError as e:
+        func_logger(
+            'Failed accessing response schema for %s request to `%s`; is the endpoint documented? ' 'Error: %s',
+            method,
+            path,
+            e,
+        )
+        return
+
+    try:
+        # validate response data with respect to response schema
+        from django_swagger_tester.schema_tester import SchemaTester
+
+        SchemaTester(
+            schema=response_schema,
+            data=response.data,
+            case_tester=settings.case_tester,
+            camel_case_parser=settings.camel_case_parser,
+            origin='response',
+        )
+        logger.info('Response valid for %s request to %s', method, path)
+    except SwaggerDocumentationError as e:
+        long_message = format_response_tester_error(e, hint=e.response_hint, addon='')
+        func_logger('Bad response returned for %s request to %s. Error: %s', method, path, str(long_message))
+    except CaseError as e:
+        func_logger('Found incorrectly cased cased key, `%s` in %s', e.key, e.origin)
+
+
+def copy_response(response: Response) -> Response:
+    """
+    Loads response data as JSON and returns a copied response object.
+    """
+    # By parsing the response data JSON we bypass problems like uuid's not having been converted to
+    # strings yet, which otherwise would create problems when comparing response data types to the
+    # documented schema types in the schema tester
+    content = response.content.decode(response.charset)
+    response_data = json.loads(content)
+    copied_response = deepcopy(response)
+    copied_response.data = response_data  # this can probably be done differently
```

Comment:
Not sure why it needs to be done differently? This looks clean to me. 
I’d remove the comment and just return it here.